src/github.com/matrix-org/dendrite/roomserver/storage/storage.go

```diff
@@ -380,6 +382,27 @@ func (d *Database) StateEntriesForTuples(
 	return d.statements.bulkSelectFilteredStateBlockEntries(stateBlockNIDs, stateKeyTuples)
 }
 
+// MembershipUpdater implements input.RoomEventDatabase
+func (d *Database) MembershipUpdater(roomID, targetUserID string) (updater types.MembershipUpdater, err error) {
+	err = common.WithTransaction(d.db, func(txn *sql.Tx) error {
```

Comment:
erm; will this not commit the transaction being returned in `updater.txn`, this making it a bit pointless?